python/rapidsmpf/rapidsmpf/config.pyx

```diff
@@ -87,73 +38,130 @@ cdef class Options:
         with nogil:
             self._handle = cpp_Options()
 
-    def get_or_assign(self, str key, parser_type, default_value):
+    def get(self, str key, *, return_type, factory):
         """
-        Get the value of the given key, or assign and return a default value.
+        Retrieves a configuration option by key.
 
-        When a key is accessed for the first time, its option value (as a string)
-        is parsed using the `parser_type`. If the key has no option value,
-        `default_value` is assigned to the key and returned.
+        If the option is not present, it is constructed using the provided
+        factory function, which receives the string representation of the
+        option (or an empty string if unset). The option is cached after
+        the first access.
 
-        The type of the returned value is determined by the `parser_type` argument,
-        which must be one of the supported Python types: bool, int, float, or str.
-        The `default_value` is cast to the corresponding C++ type before being passed
-        to the underlying options system.
+        The option is cast to the specified `return_type`, which must be one
```

Comment:
This should probably be 

```suggestion
        The option is cast to the specified ``return_type``, which must be one
```

I'm guessing this isn't included in our docs yet. Probably just need to add it to the index at https://github.com/rapidsai/rapidsmpf/blob/branch-25.06/docs/source/index.md?plain=1. LMK if you want to address that here & need help with it.

Likewise on L74 and L126 below.